src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -614,7 +665,11 @@ private String encodePathSegment(final String pathSegment) {
     return xs;
   }
 
-  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url, final String username, final String password, final SSLConnectionSocketFactory sslConnectionSocketFactory, final SSLContext sslContext) throws MalformedURLException {
+  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
+                                                                   final String username, final String password,
+                                                                   final SSLConnectionSocketFactory sslConnectionSocketFactory,
+                                                                   final SSLContext sslContext,
+                                                                   final HttpClientBuilderConfigurator configurator) throws MalformedURLException {
```

Comment:
I don't see how I can easily replace `RestTemplate#getForObject` with a version that accepts an `HttpEntity`.

So I'll inline the default implementation and merge.